subsys/emds/emds.c

```diff
@@ -11,60 +11,68 @@
 #include <drivers/flash.h>
 #include "emds_flash.h"
 
-#if defined(CONFIG_BT) && !defined(CONFIG_BT_LL_SW_SPLIT)
-#include <sdc.h>
-#include <mpsl.h>
-#endif
-
 #include <logging/log.h>
 LOG_MODULE_REGISTER(emds, CONFIG_EMDS_LOG_LEVEL);
 
 K_SEM_DEFINE(emds_sem, 0, 1);
 static bool emds_ready;
 static bool emds_initialized;
 static bool emds_complete;
+static uint32_t store_key;
 
 static sys_slist_t emds_dynamic_entries;
 static struct emds_fs emds_flash;
+static emds_store_cb_t app_store_cb;
 
 static void emds_handler(void)
 {
-	k_sem_reset(&emds_sem);
-	k_sem_take(&emds_sem, K_FOREVER);
+	while (true) {
```

Comment:
Just courious: Could you elaborate why you have encapsulated this in a while loop? Is there a situation where we can allow successive stores without reboot in between?

Reply:
Yes, by doing a prepare between them.